.github/workflows/tests.yaml

```
@@ -0,0 +1,29 @@
+name: Unit Tests
+on: [push, pull_request_target]
+
+jobs:
+  tests:
+    name: Tests (Go ${{ matrix.go-version }})
+    runs-on: ubuntu-latest
+    # always run on push events
+    # only run on pull_request_target event when pull request pulls from fork repository
+    if: >
+      github.event_name == 'push' || 
+      github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.full_name != github.repository 
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [1.11, 1.12, 1.13, 1.14, 1.15, "^1"]
```

[Go only supports the last two major releases](https://golang.org/doc/devel/release.html#:~:text=checkout%20goX.Y.Z-,Release%20Policy,until%20the%20Go%201.8%20release.&text=1%2C%20Go%201.6.,2%2C%20and%20so%20on).), so I think we're fine only running it against a few latest ones here (and we can add 1.16 too):
```suggestion
        go-version: [1.14, 1.15, 1.16, "^1"]
```